{
  "gsd": {
    "metadata": {
      "exploitCode": "unknown",
      "remediation": "unknown",
      "reportConfidence": "confirmed",
      "type": "vulnerability"
    },
    "osvSchema": {
      "aliases": [
        "CVE-2024-0450"
      ],
      "details": "An issue was found in the CPython `zipfile` module affecting versions 3.12.2, 3.11.8, 3.10.13, 3.9.18, and 3.8.18 and prior.\n\nThe zipfile module is vulnerable to \u201cquoted-overlap\u201d zip-bombs which exploit the zip format to create a zip-bomb with a high compression ratio. The fixed versions of CPython makes the zipfile module reject zip archives which overlap entries in the archive.\n\n",
      "id": "GSD-2024-0450",
      "modified": "2024-01-12T06:02:16.909485Z",
      "schema_version": "1.4.0"
    }
  },
  "namespaces": {
    "cve.org": {
      "CVE_data_meta": {
        "ASSIGNER": "cna@python.org",
        "ID": "CVE-2024-0450",
        "STATE": "PUBLIC"
      },
      "affects": {
        "vendor": {
          "vendor_data": [
            {
              "product": {
                "product_data": [
                  {
                    "product_name": "CPython",
                    "version": {
                      "version_data": [
                        {
                          "version_affected": "<=",
                          "version_name": "3.12.0",
                          "version_value": "3.12.2"
                        },
                        {
                          "version_affected": "<=",
                          "version_name": "3.11.0",
                          "version_value": "3.11.8"
                        },
                        {
                          "version_affected": "<=",
                          "version_name": "3.10.0",
                          "version_value": "3.10.13"
                        },
                        {
                          "version_affected": "<=",
                          "version_name": "3.9.0",
                          "version_value": "3.9.18"
                        },
                        {
                          "version_affected": "<=",
                          "version_name": "0",
                          "version_value": "3.8.18"
                        }
                      ]
                    }
                  }
                ]
              },
              "vendor_name": "Python Software Foundation"
            }
          ]
        }
      },
      "data_format": "MITRE",
      "data_type": "CVE",
      "data_version": "4.0",
      "description": {
        "description_data": [
          {
            "lang": "eng",
            "value": "An issue was found in the CPython `zipfile` module affecting versions 3.12.2, 3.11.8, 3.10.13, 3.9.18, and 3.8.18 and prior.\n\nThe zipfile module is vulnerable to \u201cquoted-overlap\u201d zip-bombs which exploit the zip format to create a zip-bomb with a high compression ratio. The fixed versions of CPython makes the zipfile module reject zip archives which overlap entries in the archive.\n\n"
          }
        ]
      },
      "generator": {
        "engine": "Vulnogram 0.1.0-dev"
      },
      "impact": {
        "cvss": [
          {
            "attackComplexity": "LOW",
            "attackVector": "LOCAL",
            "availabilityImpact": "HIGH",
            "baseScore": 6.2,
            "baseSeverity": "MEDIUM",
            "confidentialityImpact": "NONE",
            "integrityImpact": "NONE",
            "privilegesRequired": "NONE",
            "scope": "UNCHANGED",
            "userInteraction": "NONE",
            "vectorString": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
            "version": "3.1"
          }
        ]
      },
      "problemtype": {
        "problemtype_data": [
          {
            "description": [
              {
                "cweId": "CWE-405",
                "lang": "eng",
                "value": "CWE-405"
              }
            ]
          }
        ]
      },
      "references": {
        "reference_data": [
          {
            "name": "https://github.com/python/cpython/commit/66363b9a7b9fe7c99eba3a185b74c5fdbf842eba",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/commit/66363b9a7b9fe7c99eba3a185b74c5fdbf842eba"
          },
          {
            "name": "https://github.com/python/cpython/commit/fa181fcf2156f703347b03a3b1966ce47be8ab3b",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/commit/fa181fcf2156f703347b03a3b1966ce47be8ab3b"
          },
          {
            "name": "https://github.com/python/cpython/commit/a956e510f6336d5ae111ba429a61c3ade30a7549",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/commit/a956e510f6336d5ae111ba429a61c3ade30a7549"
          },
          {
            "name": "https://github.com/python/cpython/commit/30fe5d853b56138dbec62432d370a1f99409fc85",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/commit/30fe5d853b56138dbec62432d370a1f99409fc85"
          },
          {
            "name": "https://github.com/python/cpython/commit/a2c59992e9e8d35baba9695eb186ad6c6ff85c51",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/commit/a2c59992e9e8d35baba9695eb186ad6c6ff85c51"
          },
          {
            "name": "https://github.com/python/cpython/commit/d05bac0b74153beb541b88b4fca33bf053990183",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/commit/d05bac0b74153beb541b88b4fca33bf053990183"
          },
          {
            "name": "https://github.com/python/cpython/issues/109858",
            "refsource": "MISC",
            "url": "https://github.com/python/cpython/issues/109858"
          },
          {
            "name": "https://www.bamsoftware.com/hacks/zipbomb/",
            "refsource": "MISC",
            "url": "https://www.bamsoftware.com/hacks/zipbomb/"
          },
          {
            "name": "https://mail.python.org/archives/list/security-announce@python.org/thread/XELNUX2L3IOHBTFU7RQHCY6OUVEWZ2FG/",
            "refsource": "MISC",
            "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/XELNUX2L3IOHBTFU7RQHCY6OUVEWZ2FG/"
          },
          {
            "name": "https://lists.debian.org/debian-lts-announce/2024/03/msg00024.html",
            "refsource": "MISC",
            "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00024.html"
          },
          {
            "name": "https://lists.debian.org/debian-lts-announce/2024/03/msg00025.html",
            "refsource": "MISC",
            "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00025.html"
          }
        ]
      },
      "source": {
        "discovery": "UNKNOWN"
      }
    },
    "nvd.nist.gov": {
      "cve": {
        "descriptions": [
          {
            "lang": "en",
            "value": "An issue was found in the CPython `zipfile` module affecting versions 3.12.2, 3.11.8, 3.10.13, 3.9.18, and 3.8.18 and prior.\n\nThe zipfile module is vulnerable to \u201cquoted-overlap\u201d zip-bombs which exploit the zip format to create a zip-bomb with a high compression ratio. The fixed versions of CPython makes the zipfile module reject zip archives which overlap entries in the archive.\n\n"
          },
          {
            "lang": "es",
            "value": "Se encontr\u00f3 un problema en el m\u00f3dulo `zipfile` de CPython que afecta a las versiones 3.12.2, 3.11.8, 3.10.13, 3.9.18 y 3.8.18 y anteriores. El m\u00f3dulo zipfile es vulnerable a bombas zip \"superpuestas entre comillas\" que explotan el formato zip para crear una bomba zip con una alta relaci\u00f3n de compresi\u00f3n. Las versiones fijas de CPython hacen que el m\u00f3dulo zipfile rechace archivos zip que se superponen con entradas en el archivo."
          }
        ],
        "id": "CVE-2024-0450",
        "lastModified": "2024-03-24T23:15:09.027",
        "metrics": {
          "cvssMetricV31": [
            {
              "cvssData": {
                "attackComplexity": "LOW",
                "attackVector": "LOCAL",
                "availabilityImpact": "HIGH",
                "baseScore": 6.2,
                "baseSeverity": "MEDIUM",
                "confidentialityImpact": "NONE",
                "integrityImpact": "NONE",
                "privilegesRequired": "NONE",
                "scope": "UNCHANGED",
                "userInteraction": "NONE",
                "vectorString": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                "version": "3.1"
              },
              "exploitabilityScore": 2.5,
              "impactScore": 3.6,
              "source": "cna@python.org",
              "type": "Secondary"
            }
          ]
        },
        "published": "2024-03-19T16:15:09.180",
        "references": [
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/commit/30fe5d853b56138dbec62432d370a1f99409fc85"
          },
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/commit/66363b9a7b9fe7c99eba3a185b74c5fdbf842eba"
          },
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/commit/a2c59992e9e8d35baba9695eb186ad6c6ff85c51"
          },
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/commit/a956e510f6336d5ae111ba429a61c3ade30a7549"
          },
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/commit/d05bac0b74153beb541b88b4fca33bf053990183"
          },
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/commit/fa181fcf2156f703347b03a3b1966ce47be8ab3b"
          },
          {
            "source": "cna@python.org",
            "url": "https://github.com/python/cpython/issues/109858"
          },
          {
            "source": "cna@python.org",
            "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00024.html"
          },
          {
            "source": "cna@python.org",
            "url": "https://lists.debian.org/debian-lts-announce/2024/03/msg00025.html"
          },
          {
            "source": "cna@python.org",
            "url": "https://mail.python.org/archives/list/security-announce@python.org/thread/XELNUX2L3IOHBTFU7RQHCY6OUVEWZ2FG/"
          },
          {
            "source": "cna@python.org",
            "url": "https://www.bamsoftware.com/hacks/zipbomb/"
          }
        ],
        "sourceIdentifier": "cna@python.org",
        "vulnStatus": "Awaiting Analysis",
        "weaknesses": [
          {
            "description": [
              {
                "lang": "en",
                "value": "CWE-405"
              }
            ],
            "source": "cna@python.org",
            "type": "Secondary"
          }
        ]
      }
    }
  }
}